### PR TITLE
filters: use importlib_resources API to avoid deprecation warning

### PR DESCRIPTION
**Pull request by polariton**
Fix deprecation warning "pkg_resources is deprecated as an API"

### CODE REVIEW

**Comment by bmcfee** [comment]
Thanks - looks like you need to add `importlib_resources` to the dependencies to get CI working though.

**Comment by polariton** [comment]
OK, added importlib_resources to CI.

**Comment by bmcfee** [comment]
This should be done via package requirements, not by changing the test environment script.

**Review by avalentino — APPROVED** [review]
LGTM

**Comment by bmcfee** [comment]
Sorry, this fell off my radar.  It looks like the tests are failing with this PR, but I'm not sure why they're not showing up in the GH actions checks here.

**Comment by bmcfee** [comment]
https://pipelinesghubeus26.actions.githubusercontent.com/bUxMb2yqtuvB2RTCARL2QpDTvLpo5Yf3UEj1GdiJq0K3dNEhMw/_apis/pipelines/1/runs/8/signedlogcontent/3?urlExpires=2024-03-04T19%3A26%3A50.4934762Z&urlSigningMethod=HMACV1&urlSignature=wFq6a%2F9ay5mg0BrMgePItMgMpWkGm0bjH2TKOSBowcM%3D is the failed build log on 3.6.  (Not that I'm too fussed with maintaining for 3.6, but if we're going to put out a release to support newer python, we shouldn't break the old version if we don't have to.)

Failures on 3.10+ environments were due to codecov being flakey and nothing with our test suite directly.

**Comment by takenori-y** [comment]
I could not see the log, but the variable `__name__` seems to be different in python 3.6 and 3.7 and later.
How is this?

```py
if filter_name not in FILTER_CACHE:
    package = __name__.split(".")[0]  # or simply 'resampy'
    fname = importlib_resources.files(package).joinpath(
        'data', os.path.extsep.join([filter_name, 'npz'])
    )
```

File: resampy/filters.py [hunk 1]
```diff
@@ -203,10 +202,10 @@ def load_filter(filter_name):
     '''
 
     if filter_name not in FILTER_CACHE:
-        fname = os.path.join('data',
-                             os.path.extsep.join([filter_name, 'npz']))
+        fname = importlib_resources.files(__name__) / 'data' / f'{filter_name}.npz'
```
Comment (bmcfee) [inline resampy/filters.py:205]:
```suggestion
        fname = importlib_resources.files("resampy") / 'data' / f'{filter_name}.npz'
```

**Comment by bmcfee** [comment]
I just did the simple thing and hacked the package name directly in.  Let's see if this makes CI happy.

**Comment by bmcfee** [comment]
Alright, good enough.  I think we still need to do a bit of tweaking on the package setup.cfg, but this is functional for now.  I'll try to put a release out this week.

**Pull request merged**